bindings/matrix-sdk-ffi/src/platform.rs

```diff
@@ -94,36 +98,126 @@ fn text_layers<S>(config: TracingConfiguration) -> impl Layer<S>
 where
     S: Subscriber + for<'a> LookupSpan<'a>,
 {
-    fn fmt_layer<S>() -> fmt::Layer<S> {
-        fmt::layer()
-            .with_file(true)
-            .with_line_number(true)
-            .with_ansi(cfg!(not(any(target_os = "android", target_os = "ios"))))
+    // Adjusted version of tracing_subscriber::fmt::Format
+    struct EventFormatter {
+        display_timestamp: bool,
+        display_level: bool,
     }
 
-    let file_layer = config
-        .write_to_files
-        .map(|c| fmt_layer().with_writer(tracing_appender::rolling::hourly(c.path, c.file_prefix)));
+    impl EventFormatter {
+        fn new() -> Self {
+            Self { display_timestamp: true, display_level: true }
+        }
 
-    #[cfg(not(target_os = "android"))]
-    return Layer::and_then(
-        file_layer,
-        config.write_to_stdout_or_system.then(|| fmt_layer().with_writer(std::io::stderr)),
-    );
+        #[cfg(target_os = "android")]
+        fn for_logcat() -> Self {
+            // Level and time are already captured by logcat separately
+            Self { display_timestamp: false, display_level: false }
+        }
 
-    #[cfg(target_os = "android")]
-    return Layer::and_then(
+        fn format_timestamp(&self, writer: &mut fmt::format::Writer<'_>) -> std::fmt::Result {
+            if fmt::time::SystemTime.format_time(writer).is_err() {
+                writer.write_str("<unknown time>")?;
+            }
+            Ok(())
+        }
+
+        fn write_filename(
+            &self,
+            writer: &mut fmt::format::Writer<'_>,
+            filename: &str,
+        ) -> std::fmt::Result {
+            const CRATES_IO_PATH_MATCHER: &str = ".cargo/registry/src/index.crates.io";
+            let crates_io_filename = filename
+                .split_once(CRATES_IO_PATH_MATCHER)
+                .and_then(|(_, rest)| rest.split_once('/').map(|(_, rest)| rest));
```

Comment:
Could that be done with `trim_left_matches` or `strip_prefix` instead of split_once + skip the left part?

Reply:
No, because it's not a prefix in both cases. It's a part of the string somewhere in the middle. (the `CRATES_IO_PATH_MATCHER` intentionally does not contain a trailing `/` because there's a `-hash` at the end of the last directory name)

Reply:
Alrighty then :+1: